community/FixEarthLighting.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: page
+title: Fixing lighting issues with Chunky
+parent: Community Guides
+nav_order: 10
+---
+
+Earth maps are usually created using third party tools such as WorldPainter, which create minimal chunk data, often missing important components such as lighting information.
+Normally, you won't notice this, because the server will automatically fix lighting data when players load the chunks.
+However, BlueMap directly reads chunk data from disk. This is much more efficient and allows BlueMap to easily support many platforms, but also means the chunk data needs to be complete before BlueMap can read it properly.
+But due to this, the server doesn't fix a chunk's lighting data before BlueMap renders it. This is why we need to use a tool like Chunky to fix them!
+
+So, firstly, make sure you have Chunky installed. You can download it here: [modrinth.com/chunky](https://modrinth.com/plugin/chunky)
+
+Now, to fix the lighting issues in your world:
+1. Shut down your server
+2. Open Chunky's config file: `Chunky/config.yml`
+3. Set `force-load-existing-chunks:` to `true`, and save and close the file
+4. Start your server back up again
+5. Run this command: `/chunky radius 1000k`
+    - Or use an even bigger number. Just make sure that your whole world is enveloped by this.
+    - This sets Chunky to (re)generate a million radius (But don't worry! It won't _actually_ do that, due to the next step!)
+6. Run this command: `/chunky pattern world`
+   - This sets Chunky to only go over and fix the chunks that have already been generated. This prevents it from going all the way out to the 1000k blocks!
+7. Run this command: `/chunky start`
+    - This starts the cleanup process. This may take a while, so be patient!
+8. Once it is done, **restart your server!**
+    - This is very important, to force Minecraft to acknowledge the fixed chunks.
+9. And finally run `/bluemap purge <map-id>` to rerender your world! This time with all cleaned-up lighting :D
+
+If you want a border around your map and if you got your Earth map from [earth.motfe.net](https://earth.motfe.net/), then you can install ChunkyBorder: [modrinth.com/chunkyborder](https://modrinth.com/plugin/chunkyborder).  
+You can then find the exact commands and coordinates to add your border at, on this website: [docs.apocmc.us/minecraft-earth-map/map-borders](https://docs.apocmc.us/minecraft-earth-map/map-borders#id-1-500-scale-map)  
+Please note that you do _not_ have to run `/chunky start`, though, like this website suggests!
```

Comment:
```suggestion

## Chunky Instructions

First make sure you have Chunky installed. You can download it at [https://modrinth.com/plugin/chunky](https://modrinth.com/plugin/chunky).

Now to fix lighting issues in the world, we will want to force re-loading of all of the chunks in the world. This can be done using the steps outlined below.

1. Open the config file for Chunky: `Chunky/config.yml`
2. Set the option `force-load-existing-chunks` to `true`, and save the file
3. Restart the server, or run `chunky reload` to reload the configuration file on the server
4. Set the world using `/chunky world earth` (or the name of the world if not named "earth")
5. Set the radius using `/chunky radius 100000` (or large enough to cover the entire world)
6. Set the world pattern using `/chunky pattern world` which will specify re-loading only existing chunks in the world
7. Run `/chunky start` to start the loading process. This may take a long time so patience is required.
8. Restart the server when this completes. This is important to make sure all fixed chunks have been saved properly.
9. Run `/bluemap purge <map-id>` to re-render your world, this time hopefully with fixed lighting!

## Light Cleaner Instructions

First make sure you have Light Cleaner installed. You can download it at [https://www.spigotmc.org/resources/light-cleaner.42469/](https://www.spigotmc.org/resources/light-cleaner.42469/).

Now to fix lighting issues in the world, we will want to run the light cleaner.

1. Run `/cleanlight world earth` (or the name of the world if not named "earth")
2. Check `/cleanlight status` for progress. This may take a long time so patience is required.
3. Restart the server when this completes. This is important to make sure all fixed chunks have been saved properly.
4. Run `/bluemap purge <map-id>` to re-render your world, this time hopefully with fixed lighting!
```